akka-actor/src/main/scala/akka/io/WithUdpSend.scala

```diff
@@ -44,10 +46,10 @@ private[io] trait WithUdpSend {
       pendingSend = send
       pendingCommander = sender()
       if (send.target.isUnresolved) {
-        Dns.resolve(send.target.getHostName)(context.system, self) match {
+        Dns.resolve(DnsProtocol.Resolve(send.target.getHostName))(context.system, self) match {
           case Some(r) =>
             try {
-              pendingSend = pendingSend.copy(target = new InetSocketAddress(r.addr, pendingSend.target.getPort))
+              pendingSend = pendingSend.copy(target = new InetSocketAddress(r.address(), pendingSend.target.getPort))
```

Comment:
So this can now throw, was the old code `.addr` just a field access or would that throw somewhere else? (Thinking about the scope of the try-catch being potentially different here)

Reply:
It would previously have thrown as well. Old code is:

```
    val addrOption: Option[InetAddress] = IpVersionSelector.getInetAddress(ipv4.headOption, ipv6.headOption)

    @throws[UnknownHostException]
    def addr: InetAddress = addrOption match {
      case Some(ipAddress) => ipAddress
      case None            => throw new UnknownHostException(name)
    }
```

Reply:
Not to say that is handled correctly, just that is hasn't changed in this PR

Reply:
Thanks for checking! All good as is then.